File: trunk/src/dotnet/yaxgl/YaxglLibrary/SimpleComponentFactory.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;

namespace de.yaxgl
{
    public class SimpleComponentFactory
    {

        class NoImplementationForXmlElementException : Exception 
        {
            public NoImplementationForXmlElementException()
            {
            
            }
            public NoImplementationForXmlElementException(string message)
                : base(message)
            {
            
            }
            
            public NoImplementationForXmlElementException(string message, Exception inner)
            : base(message, inner)
            {
            
            }
        }

        public static Containable createComponent(Container owner,XmlElement xmlElement)
        {
            Containable component=null;
            
            if (xmlElement.LocalName.Equals("button"))
            {
                component = new Button(owner, xmlElement.Attributes["id"].InnerText);
            }            
            else if (xmlElement.LocalName.Equals("label"))
            {
                component = new Label(owner, xmlElement.Attributes["id"].InnerText);                
            }
            else if (xmlElement.LocalName.Equals("editbox"))
            {
                component = new EditBox(owner, xmlElement.Attributes["id"].InnerText);                
            }
            else if (xmlElement.LocalName.Equals("textbox"))
            {
                component = new TextBox(owner, xmlElement.Attributes["id"].InnerText);
             }
            else if (xmlElement.LocalName.Equals("checkbox"))
            {
                component = new CheckBox(owner, xmlElement.Attributes["id"].InnerText);                
            }
            else if (xmlElement.LocalName.Equals("imagebox"))
            {
                component = new ImageBox(owner, xmlElement.Attributes["id"].InnerText);
            }
            else if (xmlElement.LocalName.Equals("combobox"))
            {
                component = new ComboBox(owner, xmlElement.Attributes["id"].InnerText);                
            }
            else if (xmlElement.LocalName.Equals("listbox"))
            {
                component = new ListBox(owner, xmlElement.Attributes["id"].InnerText);                
            }
            else if (xmlElement.LocalName.Equals("radiobutton"))
            {
                component = new RadioButton(owner, xmlElement.Attributes["id"].InnerText);
            }
            else if (xmlElement.LocalName.Equals("groupref"))
            {
                component = new Group(xmlElement.Attributes["source"].InnerText, owner, xmlElement.Attributes["id"].InnerText);
            }
            else if (xmlElement.LocalName.Equals("groupbox"))
            {
                component = new GroupBox(xmlElement, owner, xmlElement.Attributes["id"].InnerText);
            }
            else if (xmlElement.LocalName.Equals("radiogroup"))
            {
                component = new RadioGroup(xmlElement,owner, xmlElement.Attributes["id"].InnerText);
            }
            else
            {
                throw new NoImplementationForXmlElementException("Error on creating component: No implementation for "+xmlElement.LocalName+ " jet available in this library." );
            }

            ((Component)component).initializeNativeControl(xmlElement);
            return component;
        }
    
    }
}

File: src/java/YaxglLibrary/de/yaxgl/SimpleComponentFactory.java
package de.yaxgl;


import org.w3c.dom.Element;

import de.yaxgl.Base.*;
import de.yaxgl.Container.*;
import de.yaxgl.Controls.*;


public class SimpleComponentFactory {
	  
	@SuppressWarnings("serial")
	public class NoImplementationForXmlElementException extends Exception
      {
          public NoImplementationForXmlElementException()
          {

          }
          public NoImplementationForXmlElementException(String message)
          {
        	  super(message);
          }

          public NoImplementationForXmlElementException(String message, Exception inner)
          {
        	  super(message,inner);
          }
      }

      public static Containable createComponent(Container owner, Element xmlElement) 
      {
          Containable component = null;

          //TODO: the thing with the namespce uri and the localElementName
          if (xmlElement.getNodeName().equals("yaxgl:button"))
          {
              component = new Button(owner, xmlElement.getAttribute("id"));
          }
          else if (xmlElement.getNodeName().equals("yaxgl:label"))
          {
              component = new Label(owner, xmlElement.getAttribute("id"));
          }
          else if (xmlElement.getNodeName().equals("yaxgl:editbox"))
          {
              component = new EditBox(owner, xmlElement.getAttribute("id"));
          }
          else if (xmlElement.getNodeName().equals("yaxgl:textbox"))
          {
              component = new TextBox(owner, xmlElement.getAttribute("id"));
          }
          else if (xmlElement.getNodeName().equals("yaxgl:checkbox"))
          {
              component = new CheckBox(owner, xmlElement.getAttribute("id"));
          }
          else if (xmlElement.getNodeName().equals("yaxgl:imagebox"))
          {
              component = new ImageBox(owner, xmlElement.getAttribute("id"));
          }
          else if (xmlElement.getNodeName().equals("yaxgl:combobox"))
          {
              component = new ComboBox(owner, xmlElement.getAttribute("id"));
          }
          else if (xmlElement.getNodeName().equals("yaxgl:listbox"))
          {
             // component = new ListBox(owner, xmlElement.getAttribute("id"));
          }
          else if (xmlElement.getNodeName().equals("yaxgl:radiobutton"))
          {
             // component = new RadioButton(owner, xmlElement.getAttribute("id"));
          }
          else if (xmlElement.getNodeName().equals("yaxgl:groupref"))
          {
             // component = new Group(xmlElement.getAttribute("source"), owner, xmlElement.getAttribute("id"));
          }
          else if (xmlElement.getNodeName().equals("yaxgl:groupbox"))
          {
            //  component = new GroupBox(xmlElement, owner, xmlElement.getAttribute("id"));
          }
          else if (xmlElement.getNodeName().equals("yaxgl:radiogroup"))
          {
             // component = new RadioGroup(xmlElement, owner, xmlElement.getAttribute("id"));
          }
          else
          {
             // throw new NoImplementationForXmlElementException("Error on creating component: No implementation for " + xmlElement.getLocalName() + " jet available in this library.");
			
          }

          /*init the component with the in xml given attributes*/
          ((Component)component).initializeNativeControl(xmlElement);
          return component;
      }
}
